crates/ruff_formatter/src/format_element.rs

```diff
@@ -47,6 +47,13 @@ pub enum FormatElement {
         slice: SyntaxTokenText,
     },
 
+    /// Token constructed by slicing a defined range from a static string
+    StaticTextSlice {
+        text: Rc<str>,
+        start: usize,
+        end: usize,
```

Comment:
I recommend you to use `TextRange` here. Documents larger than 4GB should be rare. Using `TextRange` helps keeping the overall size of `FormatElement` low which is essential for good formatter performance (fewer bytes to write when formatting, fewer bytes to read when printing, better cache locality...) and memory usage low.